lib/BackbeatConsumer.js

```diff
@@ -148,10 +149,19 @@ class BackbeatConsumer extends EventEmitter {
         /** @type {ConsumerStats} */
         this.consumerStats = { lag: {} };
 
+        /** @type {Kafka.TopicPartition[]} */
+        this._assignment = [];
+        /** @type {Map<string,Kafka.TopicPartition[]>} */
+        // Key format is "topic:partition"
+        this._pausedPartitions = new Map();
+
         this._circuitBreaker = new CircuitBreaker(circuitBreaker);
         this._circuitBreakerMetrics = circuitBreakerMetrics;
+        this._circuitBreaker.on('state-changed', this._onCircuitBreakerStateChanged.bind(this));
 
-        this._init();
+        if (init) {
```

Comment:
not found of this flag, it hints that there is a design flaw here.

can we find a better to test (so we don't need this flag), and/or a different way to integrate the code so that the "pause/resume" logic can be unit tested separately (since this is what you try to test with this flag, I gather)

Reply:
note: this (kind of ugly) hack may do the trick...

```
consumer = new BackbeatConsumer({bootstrap: true, ...});
consumer._bootstrapping = true;
```

Reply:
another apporach may be to introduce a wrapper/proxy for KafkaConsumer, which can change internal state depending on circuit breaker + wrap the assign/unassign methods to pause/resume when needed